### PR TITLE
ui/driverview: hide on interactive timeout

### PR DESCRIPTION
**Pull request by deanlee**
fis issue https://github.com/commaai/openpilot/pull/27457#issuecomment-1681631826:

> after https://github.com/commaai/openpilot/pull/28656, when it does time out, the driver view stays up and tapping once doesn't close it. If there's a signal that gets emitted when the timeout happens, we should close the driver view then.

### CODE REVIEW

File: selfdrive/ui/qt/offroad/driverview.cc [hunk 1]
```diff
@@ -19,11 +19,19 @@ DriverViewWindow::DriverViewWindow(QWidget* parent) : QWidget(parent) {
   connect(cameraView, &CameraWidget::vipcThreadFrameReceived, scene, &DriverViewScene::frameUpdated);
   layout->addWidget(scene);
   layout->setCurrentWidget(scene);
+
+  QObject::connect(device(), &Device::interactiveTimeout, this, &DriverViewWindow::requestDone);
+}
+
+void DriverViewWindow::requestDone() {
```
Comment (sshane) [inline selfdrive/ui/qt/offroad/driverview.cc:26]:
```suggestion
void DriverViewWindow::closeView() {
```
?

**Comment by sshane** [comment]
I assume `hideEvent` isn't fired when the screen turns off since it stays up, right?

**Comment by deanlee** [comment]
> I assume `hideEvent` isn't fired when the screen turns off since it stays up, right?

yes

**Comment by sshane** [comment]
when the 60s timer runs out, it sets the timeout to 15s, so the screen turns off after 60+30s

**Comment by deanlee** [comment]
think we should not call resetInteractiveTimeout in hideEvent. just let MainWindow::eventFilter reset it on clicks.

**Pull request merged**